crates/stc_ts_type_checker/tests/conformance/types/tuple/arityAndOrderCompatibility01.ts

```diff
@@ -33,3 +33,10 @@ var n3: [number, string] = z;
 var o1: [string, number] = x;
 var o2: [string, number] = y;
 var o3: [string, number] = y;
+
+x = y;
+// x = z; // should get TS2322 but pass
+y = x;
+y = z;
+// z = x; // should pass but got TS2322
```

Comment:
According to TS Playground,
It seems that `x = z` and `z = x` work in opposite directions.
Should I follow TS playground and fix it in this PR?
![image](https://github.com/dudykr/stc/assets/51254761/38a250e1-0edc-41ee-abc8-9952a04d5a73)

Reply:
I think that this issue is related to the following issue @kdy1 
#1005

Reply:
@sunrabbit123
Thanks for the information.
So it looks like it does not need to be fixed, at least in this PR.